unitary/alpha/quantum_world.py

```diff
@@ -526,6 +557,103 @@ def get_binary_probabilities(
             binary_probs.append(1 - one_probs[0])
         return binary_probs
 
+    # TODO(pengfeichen): make this function work for compile_to_qubits==True.
+    def get_binary_probabilities_from_state_vector(
+        self, objects: Optional[Sequence[QuantumObject]] = None, decimals: int = 2
+    ) -> List[float]:
+        """Calculates the total probabilities for all non-zero states
+        based on simulating the state vector.
+
+        Parameters:
+            objects:    List of QuantumObjects
+            decimals:   Number of rounding decimals of the real and imaginary coefficients
+                        of the state vector.
+
+        Returns:
+            A list with one element for each object which contains the probability
+            for the event state!=0, which is the same as 1.0-Probability(state==0).
+        """
+        if objects is None:
+            objects = self.public_objects
+
+        simulator = cirq.Simulator()
```

Comment:
I don't know that this is a good idea to use cirq.Simulator().  Shouldn't we use self.sampler?

Reply:
Thanks Doug. It's updated to use self.sampler.

Reply:
Hi Doug, after some discussion this morning in the meeting, I've removed this util function. Thanks.